src/corelibs/U2View/src/ov_phyltree/TreeViewer.cpp

```diff
@@ -650,28 +668,58 @@ void TreeViewerUI::updateSettings() {
         scene()->update();
     }
 }
-void TreeViewerUI::updateTextSettings() {
-    QList<QGraphicsItem *> updatingItems = scene()->selectedItems();
-    QColor curColor = qvariant_cast<QColor>(getOptionValue(LABEL_COLOR));
-    if (updatingItems.isEmpty()) {
-        updatingItems = items();
-
-        QList<QGraphicsItem *> legendChildItems = legend->childItems();
-        if (!legendChildItems.isEmpty()) {
-            QGraphicsSimpleTextItem *legendText = dynamic_cast<QGraphicsSimpleTextItem *>(legendChildItems.first());
-            if (legendText) {
-                legendText->setBrush(curColor);
-            }
+void TreeViewerUI::getSelectedBranchSettings(QColor &color, QFont &font) const {
+    QList<QGraphicsItem *> selectedItems = scene()->selectedItems();
+    if (selectedItems.isEmpty()) {
+        return;
+    }
+    int childCount = 0;
+    GraphicsBranchItem *rootItem = nullptr;
+    for (QGraphicsItem *graphItem : qAsConst(selectedItems)) {
+        auto *branchItem = dynamic_cast<GraphicsBranchItem *>(graphItem);
```

Comment:
Do you really want to 'return' here is cast is failed? In this case it should be SAFE_POINT
or use
```
if (auto *branchItem = dynamic_cast<GraphicsBranchItem *>(graphItem)) {
   // do something with branchItem
}
```